streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -689,14 +689,12 @@ public synchronized final TopologyBuilder addSink(final String name, final Strin
         if (nodeFactories.containsKey(name))
             throw new TopologyBuilderException("Processor " + name + " is already added.");
 
-        if (parentNames != null) {
-            for (String parent : parentNames) {
-                if (parent.equals(name)) {
-                    throw new TopologyBuilderException("Processor " + name + " cannot be a parent of itself.");
-                }
-                if (!nodeFactories.containsKey(parent)) {
-                    throw new TopologyBuilderException("Parent processor " + parent + " is not added yet.");
-                }
+        for (String parent : parentNames) {
```

Comment:
Might be excessive to add `final` everywhere. I'd say if it's self-evident that things aren't mutated I could skip final.